src/x/serialize/decoder_fast.go

```diff
@@ -31,48 +31,71 @@ func TagValueFromEncodedTagsFast(
 	encodedTags []byte,
 	tagName []byte,
 ) ([]byte, bool, error) {
+	var (
```

Comment:
As we are changing the implementation of this function with ambitious name (`Fast`), I believe we also need a benchmark for it to make sure it does not regress noticeably (or maybe there already is an existing benchmark for it?).

Reply:
https://github.com/m3db/m3/pull/3087/commits/c4296b65fd467cac4df36021059c53865ef1ca05